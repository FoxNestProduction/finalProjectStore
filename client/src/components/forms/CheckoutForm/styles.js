export const subtitle = {
  mb: '30px',
  color: 'text.primary',
  fontWeight: 'fontWeightMedium',
  fontSize: {
    mobile: '20px',
    tablet: '24px',
    desktop: '26px',
  },
};

export const paymentWrapper = {
  '&.MuiFormControl-root': {
    mt: '18px',
  },
};

export const paymentRadioBtn = {
  '& .MuiTypography-root': {
    fontSize: {
      mobile: 14,
      tablet: 16,
      desktop: 18,
    },
  },
  '& .MuiSvgIcon-root': {
    fontSize: {
      tablet: 25,
    },
  },
};

export const buttonsWrapper = {
  mt: {
    mobile: '55px',
    lgTablet: '65px',
  },
  display: 'flex',
  gap: '5%',
};

export const btn = {
  p: '12px',
  borderRadius: '16px',
  width: '100%',
  fontSize: {
    tablet: '18px',
    desktop: '24px',
  },
};

export const continueBtn = {
  color: 'text.primaryLight',
  transition: 'background-color 0.3s ease, box-shadow 0.3s ease',
  ':hover': {
    backgroundColor: 'primary.hover',
  },
};

export const backBtn = {
  color: 'text.primary',
  border: 1,
  transition: 'background-color 0.3s ease, color 0.3s ease',
  ':hover': {
    backgroundColor: 'primary.hover',
    color: 'text.primaryLight',
  },
};
