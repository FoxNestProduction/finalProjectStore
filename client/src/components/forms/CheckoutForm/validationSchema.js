import { object, string } from 'yup';

const validationSchema = object({
  // firstName: string()
  //   .required('First name is required')
  //   .matches('^[A-Z][a-z]+$',
  //   'The first name must be capitalized and contain only Latin letters'),
  // lastName: string()
  //   .required('Last name is required')
  //   .matches('^[A-Z][a-z]+$',
  //   'The last name must be capitalized and contain only Latin letters'),
  name: string()
    .required('Name is required')
    .matches('^[A-Z][a-z]+$', 'The name must be capitalized and contain only Latin letters'),
  email: string()
    .required('Email is required')
    .email('Invalid email format'),
  street: string()
    .required('Enter the name of your street'),
  house: string()
    .required('Enter the number of your house'),
});

export default validationSchema;
