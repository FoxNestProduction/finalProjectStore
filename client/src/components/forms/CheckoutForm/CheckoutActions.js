import React, { memo, useCallback } from 'react';
import Button from '@mui/material/Button';
import Box from '@mui/material/Box';
import { useNavigate } from 'react-router';
import PropTypes from 'prop-types';
import { backBtn, btn, buttonsWrapper, continueBtn } from './styles';

const CheckoutActions = ({ isValid }) => {
  const navigate = useNavigate();
  const handleGoBack = useCallback(() => {
    navigate(-1);
  }, [navigate]);

  return (
    <Box sx={buttonsWrapper}>
      <Button type="button" variant="outlined" sx={{ ...btn, ...backBtn }} onClick={handleGoBack}>
        Back
      </Button>
      <Button type="submit" variant="contained" sx={{ ...btn, ...continueBtn }} disabled={!isValid}>
        Continue
      </Button>
    </Box>
  );
};

CheckoutActions.propTypes = {
  isValid: PropTypes.bool,
};

CheckoutActions.defaultProps = {
  isValid: true,
};

export default memo(CheckoutActions);
