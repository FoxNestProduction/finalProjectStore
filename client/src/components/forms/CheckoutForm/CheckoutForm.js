/* eslint-disable max-len */
import React, { memo, useEffect, useState } from 'react';
import { useNavigate } from 'react-router';
import { shallowEqual, useDispatch, useSelector } from 'react-redux';
import { Field, Form, Formik } from 'formik';

import Box from '@mui/material/Box';
import Stack from '@mui/material/Stack';
import Divider from '@mui/material/Divider';
import FormControlLabel from '@mui/material/FormControlLabel';
import MenuItem from '@mui/material/MenuItem';
import Radio from '@mui/material/Radio';
import RadioGroup from '@mui/material/RadioGroup';
import Typography from '@mui/material/Typography';
import FormControl from '@mui/material/FormControl';
import InputLabel from '@mui/material/InputLabel';
import InputMask from 'react-input-mask';

import Input from '../../inputs/Input/Input';
import validationSchema from './validationSchema';
import SelectForFormik from '../../inputs/Select/Select';
import CheckoutActions from './CheckoutActions';
import {
  subtitle,
  paymentRadioBtn, paymentWrapper,
} from './styles';
import { updateCustomer } from '../../../redux/slices/userSlice';
import { CHECKOUT_SS_KEY } from '../../../constants/constants';
import {
  getDataFromSessionStorage,
  removeDataFromSessionStorage,
  updateSessionStorageValues,
} from '../../../utils/sessionStorageHelpers';
import { putNewOrder, setPendingOrderInfo } from '../../../redux/slices/orderSlice';
import saveUserInfoToSessionStorage from '../../../utils/saveUserInfoToSessionStorage';
import { deleteCart, resetCart } from '../../../redux/slices/cartSlice';

const CheckoutForm = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const getInitialValues = () => ({
    name: '',
    email: '',
    tel: '',
    city: 'Kyiv',
    street: '',
    house: '',
    apartment: '',
    payment: 'Card',
  });

  const [initialValues, setInitialValues] = useState(getInitialValues);

  const isUserAuthorized = useSelector((state) => state.authorization.isUserAuthorized);
  const user = useSelector((state) => state.user.user, shallowEqual);
  const token = useSelector((state) => state.authorization.token);
  const cart = useSelector((state) => state.cart.cart.products, shallowEqual);

  const updateCustomerLoading = useSelector((state) => state.user.loading.updatedCustomer);
  const orderLoading = useSelector((state) => state.order.loading);
  const orderError = useSelector((state) => state.order.error);

  useEffect(() => {
    const checkoutValues = getDataFromSessionStorage(CHECKOUT_SS_KEY);
    const newValues = getInitialValues();

    if (checkoutValues) {
      Object.keys(checkoutValues).forEach((key) => {
        if (key in newValues) {
          newValues[key] = checkoutValues[key];
        }
      });
    }
    setInitialValues(newValues);
  }, [isUserAuthorized]);

  const handleFieldBlur = (e, handleBlur) => {
    handleBlur(e);
    updateSessionStorageValues(CHECKOUT_SS_KEY, { [e.target.name]: e.target.value });
  };

  const handleContinue = async (values) => {
    // updating user info in DB and user slice
    if (isUserAuthorized && token) {
      const customerUpdates = {
        telephone: values.tel,
      };
      await dispatch(updateCustomer(customerUpdates));
    }

    const { name, email, tel, city, street, house, apartment, payment } = values;
    const newOrder = {
      products: cart,
      name,
      email,
      mobile: tel,
      deliveryAddress: {
        city,
        street,
        house,
        apartment,
      },
      paymentInfo: payment,
    };

    if (isUserAuthorized && user) {
      const { _id: id } = user;
      newOrder.customerId = id;
    }

    if (values.payment === 'Cash') {
      newOrder.status = 'new_order/cash';
      const response = await dispatch(putNewOrder(newOrder)).unwrap();
      if (response.status === 200) {
        removeDataFromSessionStorage(CHECKOUT_SS_KEY);
        dispatch(resetCart());
        if (isUserAuthorized && user) {
          saveUserInfoToSessionStorage(user);
          dispatch(deleteCart());
        }
        navigate('/order-confirmation');
      }
    } else {
      dispatch(setPendingOrderInfo(newOrder));
      navigate('/checkout/payment');
    }
  };

  const setInitialTouched = () => {
    const values = getDataFromSessionStorage(CHECKOUT_SS_KEY);
    if (values) {
      return {
        name: 'name' in values,
        email: 'email' in values,
        tel: 'tel' in values,
        street: 'street' in values,
        house: 'house' in values,
      };
    }
    return null;
  };

  const setIsValid = (touched, errors) => {
    return !Object.keys(errors).some((key) => touched[key] === true);
  };

  return (
    <Formik
      initialValues={initialValues}
      onSubmit={handleContinue}
      validationSchema={validationSchema}
      enableReinitialize
      initialTouched={setInitialTouched()}
    >
      {({ handleBlur, touched, errors }) => (
        <Form>
          <Stack spacing={4}>
            <Divider />
            <Typography variant="h3" component="h2" align="left" sx={subtitle}>
              Personal Information
            </Typography>

            <Input
              name="name"
              id="checkout-name"
              label="Name*"
              bgColor="#FFF"
              onBlur={(e) => {
                handleFieldBlur(e, handleBlur);
              }}
            />
            <Input
              name="email"
              id="checkout-email"
              label="Email Address*"
              bgColor="#FFF"
              onBlur={(e) => { handleFieldBlur(e, handleBlur); }}
            />

            <Field name="tel">
              {({ field }) => (
                <InputMask
                  mask="+38 (099) 999-99-99"
                  {...field}
                  onBlur={(e) => { handleFieldBlur(e, handleBlur); }}
                >
                  <Input type="tel" name="tel" id="checkout-tel" bgColor="#FFF" label="Phone Number*" />
                </InputMask>
              )}
            </Field>

            <Divider />
            <Typography variant="h3" component="h2" align="left" sx={subtitle}>
              Delivery Information
            </Typography>

            <FormControl fullWidth>
              <InputLabel id="checkout-city-label">City*</InputLabel>
              <Field
                name="city"
                label="City*"
                component={SelectForFormik}
                labelId="checkout-city-label"
                id="checkout-city"
                bgColor="#FFF"
                onBlur={(e) => { handleFieldBlur(e, handleBlur); }}
              >
                <MenuItem value="Kyiv">Kyiv</MenuItem>
                <MenuItem value="Lviv">Lviv</MenuItem>
              </Field>
            </FormControl>

            <Input
              name="street"
              id="checkout-street"
              label="Street*"
              bgColor="#FFF"
              onBlur={(e) => { handleFieldBlur(e, handleBlur); }}
            />
            <Box sx={{ display: 'flex', gap: '5%' }}>
              <Input
                name="house"
                id="checkout-house"
                label="House*"
                bgColor="#FFF"
                onBlur={(e) => { handleFieldBlur(e, handleBlur); }}
              />
              <Input
                name="apartment"
                id="checkout-apartment"
                label="Apartment"
                bgColor="#FFF"
                onBlur={(e) => { handleFieldBlur(e, handleBlur); }}
              />
            </Box>

            <Divider />
            <Typography variant="h3" component="h2" align="left" sx={subtitle}>
              Payment method
            </Typography>

            <Field name="payment">
              {({ field }) => (
                <FormControl sx={paymentWrapper}>
                  <RadioGroup {...field} onBlur={(e) => { handleFieldBlur(e, handleBlur); }}>
                    <FormControlLabel value="Card" control={<Radio />} label="Card" sx={paymentRadioBtn} />
                    <FormControlLabel value="Cash" control={<Radio />} label="Cash" sx={paymentRadioBtn} />
                  </RadioGroup>
                </FormControl>
              )}
            </Field>

          </Stack>
          {orderError && (
            <Box>
              <Typography variant="body1" component="p" sx={{ color: 'text.error', mt: '15px', mb: '-15px' }}>
                {orderError}
              </Typography>
            </Box>
          )}
          <CheckoutActions
            isValid={setIsValid(touched, errors)}
            loading={(orderLoading || updateCustomerLoading) && !orderError}
          />
        </Form>
      )}
    </Formik>
  );
};

export default memo(CheckoutForm);
