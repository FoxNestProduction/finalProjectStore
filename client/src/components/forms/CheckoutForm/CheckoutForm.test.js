import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import { Provider, useDispatch } from 'react-redux';
import { useNavigate } from 'react-router';
import store from '../../../redux/store';
import CheckoutForm from './CheckoutForm';

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useNavigate: jest.fn(),
}));

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

describe('Checkout form component', () => {
  test('checkout form snapshot', () => {
    const navigateMock = jest.fn();
    useNavigate.mockReturnValue(navigateMock);
    const { asFragment } = render(
      <Provider store={store}>
        <CheckoutForm />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('try to submit empty form', async () => {
    const navigateMock = jest.fn();
    useNavigate.mockReturnValue(navigateMock);
    render(
      <Provider store={store}>
        <CheckoutForm />
      </Provider>,
    );

    const submitButton = screen.getByText('Continue');

    fireEvent.click(submitButton);

    await waitFor(() => {
      expect(screen.getByText('Name is required')).toBeInTheDocument();
    });
    await waitFor(() => {
      expect(screen.getByText('Email is required')).toBeInTheDocument();
    });
    await waitFor(() => {
      expect(screen.getByText('Phone number is required')).toBeInTheDocument();
    });
    await waitFor(() => {
      expect(screen.getByText('Enter the name of your street')).toBeInTheDocument();
    });
    await waitFor(() => {
      expect(screen.getByText('Enter the number of your house')).toBeInTheDocument();
    });
  });

  test('Submit correct form', async () => {
    const navigateMock = jest.fn();
    useNavigate.mockReturnValue(navigateMock);
    const dispatch = jest.fn();
    useDispatch.mockReturnValue(dispatch);

    render(
      <Provider store={store}>
        <CheckoutForm />
      </Provider>,
    );

    const nameInput = screen.getByLabelText('Name*');
    const emailInput = screen.getByLabelText('Email Address*');
    const phoneNumberInput = screen.getByLabelText('Phone Number*');
    const streetInput = screen.getByLabelText('Street*');
    const houseInput = screen.getByLabelText('House*');
    const apartmentInput = screen.getByLabelText('Apartment');
    const submitButton = screen.getByText('Continue');

    fireEvent.change(nameInput, { target: { value: 'Yurii' } });
    fireEvent.change(emailInput, { target: { value: 'example@example.com' } });
    fireEvent.change(phoneNumberInput, { target: { value: '+38 (011) 111-11-11' } });
    fireEvent.change(streetInput, { target: { value: 'Example Street' } });
    fireEvent.change(houseInput, { target: { value: '111' } });
    fireEvent.change(apartmentInput, { target: { value: '22' } });

    fireEvent.click(submitButton);

    await waitFor(() => {
      expect(dispatch).toHaveBeenCalledTimes(1);
    });
  });
});
