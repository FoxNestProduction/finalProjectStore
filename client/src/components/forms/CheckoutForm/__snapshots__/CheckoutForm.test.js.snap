// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout form component checkout form snapshot 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="MuiStack-root css-8mcjw0-MuiStack-root"
    >
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <h2
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignLeft css-1nzy6yt-MuiTypography-root"
      >
        Personal Information
      </h2>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="checkout-name"
        >
          Name*
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dnlmzz-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-describedby="helper-text"
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="checkout-name"
            name="name"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Name*
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          id="helper-text"
        />
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="checkout-email"
        >
          Email Address*
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dnlmzz-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-describedby="helper-text"
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="checkout-email"
            name="email"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Email Address*
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          id="helper-text"
        />
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="checkout-tel"
        >
          Phone Number*
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dnlmzz-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-describedby="helper-text"
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="checkout-tel"
            name="tel"
            type="tel"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Phone Number*
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          id="helper-text"
        />
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <h2
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignLeft css-1nzy6yt-MuiTypography-root"
      >
        Delivery Information
      </h2>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          id="checkout-city-label"
        >
          City*
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-18r46to-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="checkout-city-label checkout-city"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="checkout-city"
            role="button"
            tabindex="0"
          >
            Kyiv
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            name="city"
            tabindex="-1"
            value="Kyiv"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                City*
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="checkout-street"
        >
          Street*
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dnlmzz-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-describedby="helper-text"
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="checkout-street"
            name="street"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Street*
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          id="helper-text"
        />
      </div>
      <div
        class="MuiBox-root css-l7qti"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="checkout-house"
          >
            House*
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dnlmzz-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-describedby="helper-text"
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="checkout-house"
              name="house"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  House*
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            id="helper-text"
          />
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="checkout-apartment"
          >
            Apartment
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dnlmzz-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-describedby="helper-text"
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="checkout-apartment"
              name="apartment"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Apartment
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            id="helper-text"
          />
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <h2
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignLeft css-1nzy6yt-MuiTypography-root"
      >
        Payment method
      </h2>
      <div
        class="MuiFormControl-root css-1yqd45w-MuiFormControl-root"
      >
        <div
          class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1u6zfwf-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="payment"
                type="radio"
                value="Card"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Card
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1u6zfwf-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="payment"
                type="radio"
                value="Cash"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Cash
            </span>
          </label>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-qga54k"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1x2zd9l-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Back
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-r27lj0-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Continue
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</DocumentFragment>
`;
